### PR TITLE
move openstack and cephfs pools creation in ceph-osd

### PR DESCRIPTION
**Pull request by guits**
When deploying a large number of OSD nodes it can be an issue because of the
protection check [1] won't pass since it tries to create pools before all
OSDs are active.

The idea here is to move pools creation at the end of `ceph-osd` role.

[1] https://github.com/ceph/ceph/blob/e59258943bcfe3e52d40a59ff30df55e1e6a3865/src/mon/OSDMonitor.cc#L5673

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1578086

### CODE REVIEW

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-centos7_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-centos7_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-centos7_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-update_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-update_docker_cluster

File: group_vars/rhcs.yml.sample [hunk 1]
```diff
@@ -29,7 +29,7 @@ dummy:
 #  mimic: 13
 
 # Directory to fetch cluster fsid, keys etc...
-fetch_directory: ~/ceph-ansible-keys
+#fetch_directory: fetch/
```
[the remark this answers is not in the document]
Comment (andrewschoen) [reply group_vars/rhcs.yml.sample]: Agreed, this needs to stay ~/ceph-ansible-keys

File: group_vars/rhcs.yml.sample [hunk 2]
```diff
@@ -135,14 +135,14 @@ fetch_directory: ~/ceph-ansible-keys
 # - 'distro' means that no separate repo file will be added
 #  you will get whatever version of Ceph is included in your Linux distro.
 # 'local' means that the ceph binaries will be copied over from the local machine
-ceph_origin: repository
+#ceph_origin: "{{ 'repository' if ceph_rhcs or ceph_stable or ceph_dev or ceph_stable_uca or ceph_custom else 'dummy' }}" # backward compatibility with stable-2.2, will disappear in stable 3.1
```
Comment (andrewschoen) [inline group_vars/rhcs.yml.sample]: This shouldn't have been changed either, it needs to stay ``repository``

File: group_vars/rhcs.yml.sample [hunk 3]
```diff
@@ -135,14 +135,14 @@ fetch_directory: ~/ceph-ansible-keys
...
 #valid_ceph_origins:
 #  - repository
 #  - distro
 #  - local
 
 
-ceph_repository: rhcs
+#ceph_repository: "{{ 'community' if ceph_stable else 'rhcs' if ceph_rhcs else 'dev' if ceph_dev else 'uca' if ceph_stable_uca else 'custom' if ceph_custom else 'dummy' }}" # backward compatibility with stable-2.2, will disappear in stable 3.1
```
Comment (andrewschoen) [inline group_vars/rhcs.yml.sample]: This should be kept ``rhcs``

File: roles/ceph-defaults/defaults/main.yml [hunk 4]
```diff
@@ -524,3 +524,69 @@ rolling_update: false
 #####################
 docker_pull_retry: 3
 docker_pull_timeout: "300s"
+
+
```
Comment (leseb) [inline roles/ceph-defaults/defaults/main.yml]: please remove one empty line and keep 2

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-centos7_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-bluestore_osds_non_container

**Comment by guits** [comment]
@andrewschoen your comments have been addressed, could you review this again please?

**Pull request merged**